### PR TITLE
Add logic to run simulation validation asynchronously

### PR DESCRIPTION
**Pull request by aratz-lasa**
# What 🕵️‍♀️
This PR validates blocks asynchronously

# Why 🔑
This is for improving performance on block submissions

### CODE REVIEW

File: relay/submit.go [hunk 1]
```diff
@@ -70,11 +82,10 @@ func (rs *Relay) SubmitBlock(ctx context.Context, m *structs.MetricGroup, sbr st
 		return fmt.Errorf("failed to verify withdrawals: %w", err)
 	}
 
-	tValidateBlock := time.Now()
-	if err := rs.validateBlock(ctx, gasLimit, sbr); err != nil {
+	// wait for validations
+	if err := <-valErr; err != nil {
```
Comment (lukanus) [inline relay/submit.go:86]: select on error and context Done()? maybe?

File: relay/submit.go [hunk 2]
```diff
@@ -58,6 +58,18 @@ func (rs *Relay) SubmitBlock(ctx context.Context, m *structs.MetricGroup, sbr st
...
+	valErr := make(chan error, 1)
+	go func(ctx context.Context, gasLimit uint64, sbr structs.SubmitBlockRequest, chErr chan error) {
+		defer close(chErr)
+
+		tValidateBlock := time.Now()
+		if err := rs.validateBlock(ctx, gasLimit, sbr); err != nil {
+			chErr <- err
+			return
+		}
+		m.AppendSince(tValidateBlock, "submitBlock", "validateBlock")
```
Comment (lukanus) [inline relay/submit.go:70]: I wonder would this work properly with ObserveWithError. Like as it's async, shouldn't we call observe on that directly

File: relay/submit.go [hunk 3]
```diff
@@ -58,6 +58,18 @@ func (rs *Relay) SubmitBlock(ctx context.Context, m *structs.MetricGroup, sbr st
 	}
 	m.AppendSince(tCheckRegistration, "submitBlock", "checkRegistration")
 
+	valErr := make(chan error, 1)
+	go func(ctx context.Context, gasLimit uint64, sbr structs.SubmitBlockRequest, chErr chan error) {
+		defer close(chErr)
```
Comment (lukanus) [inline relay/submit.go:63]: 👍

File: relay/submit.go (hunk 2 above)
Comment (aratz-lasa) [reply relay/submit.go:70]: I added a lock to `MetricGroup`

**Pull request merged**